source/retryable-reads/retryable-reads.rst

```diff
@@ -0,0 +1,664 @@
+server selection error to propagate. In this case, the caller is able to infer
+that no attempt was made.
+
+2. Determining whether retry should be allowed
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A driver then determines if it should attempt to retry next.
+
+2a. When not to allow retry
+'''''''''''''''''''''''''''
+
+Drivers MUST attempt to execute the read command exactly once and allow any
+errors to propagate under any of the the following conditions:
+
+-  if retryable reads is not enabled **or**
+
+-  if the selected server does not support retryable reads **or**
+
+-  if the session in a transaction
+
+By allowing the error to propagate, the caller is able to infer that one attempt
+was made.
+
+2b. When to allow retry
+'''''''''''''''''''''''
+
+Drivers MUST only attempt to retry a read command if
+
+-  retryable reads are enabled **and**
+
+-  the selected server supports retryable reads **and**
+
+-  the first attempt yields a retryable error
+
+3. Deciding to allow retry, encountering the initial retryable error, and selecting a server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the driver decides to allow retry and the first attempt of a retryable read
+command encounters a retryable error, the driver MUST update its topology
+according to the Server Discovery and Monitoring spec (see `SDAM: Error Handling
+<https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#error-handling>`__)
+and capture this original retryable error. Drivers should then proceed with
+selecting a server for the retry attempt.
+
+3a. Selecting the server for retry
+''''''''''''''''''''''''''''''''''
+
+If the driver cannot select a server for the retry attempt or the newly selected
+server does not support retryable reads, retrying is not possible and drivers
+MUST raise the original retryable error. In both cases, the caller is able to
+infer that an attempt was made.
+
+3b. Sending an equivalent command for the second attempt
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+After server selection, a driver MUST send a valid command to the newly selected
+server that is equivalent [2]_ to the initial command sent to the first
+server. If the driver determines that the newly selected server may not be able
+to support a command equivalent to the initial command, drivers MUST NOT retry
+and MUST raise the original retryable error
+
+The above requirement can be fulfilled in one of two ways:
+
+1. During the retry attempt, the driver SHOULD recreate the command while
+   adhering to that operation's specification's server/wire version
+   requirements. If an error occurs while recreating the command, then the
+   driver MUST raise the original retryable error.
+
+   a. For example, if the wire version dips from *W*\ :sub:`0` to *W*\ :sub:`1`
```

Comment:
Is this really a sub-bullet of `1.`? As an example, it looks like it should just be a second paragraph.

Reply:
Ah, I can see that. Fixed!